flake.nix

```diff
@@ -25,16 +25,16 @@
           pact =
             final.haskell-nix.project' {
               src = ./.;
-              compiler-nix-name = "ghc962";
+              compiler-nix-name = "ghc981";
```

Comment:
This seems to be why this PR is making nix use 9.8.

Reply:
@larskuhtz Were we aiming to use ghc 9.8.1 in this PR? I thought so, but let me check my assumption.

Reply:
On the cabal build side, with PR there are builds for both, ghc-9.6 *and* ghc-9.8.

Not sure what we want on the nix side. Personally, I'd say that in case of doubt we should prefer the latest.